program/src/main/QuestionSet.vala

```diff
@@ -1,39 +1,67 @@
 using Xml;
 using Gee;
 
-public class QuestionSet : Object {
-    private int question_num; // question number
-    private double total_points; // total points possible
-    private double[] bounds;
-    private int page_num; // which page is this question on?
-    private Gee.HashMap<int, Mark> rubric_pool;
-    private Gee.ArrayList<Question> questions;
-
-    public QuestionSet(int q_num, double total_points, double[] bounds, int page_num, int num_questions) {
-        this.question_num = q_num;
-        this.total_points = total_points;
-        this.bounds = bounds;
-        this.page_num = page_num;
+
+public class QuestionSet : Object{
+    public int question_num; // question number
+    public double total_points; // total points possible
+    public double[] bounds;
+    public int page_num; // which page is this question on?
+    public Gee.HashMap<int, Mark> rubric_pool;
+    public Gee.ArrayList<Question> questions;
+    //private Question currQuestion;
+
+    public QuestionSet(int QNum, double total, double[] b, int pNum){
+        this.question_num = QNum;
+        this.total_points = total;
+        this.bounds = b;
+        this.page_num = pNum;
         this.questions = new Gee.ArrayList<Question>();
-        for(int i = 0; i < num_questions; i++) {
+        this.rubric_pool = new Gee.HashMap<int, Mark>();
+    }
+
+    // Fills question set with new empty questions - HAS to be called before grading starts but can only be called after the number of tests has been determined (aka, end of setup)
+    public void init_question_set(int num) { 
+        for(int i = 0; i < num; i++) { 
             Question q = new Question();
             this.questions.add(q);
         }
-        this.rubric_pool = new Gee.HashMap<int, Mark>();
     }
     
-    public void add_default_marks() {
+    public void addDefaultMarks() {
```

Comment:
This function and the following function aren't following proper naming conventions.

Reply:
oop, I think that this got reverted back to the old naming convention when I was fixing a merge conflict. Nice catch, I've changed the method name and all subsequent calls to match proper naming conventions.